cmd/waku/server/rest/message.go

```diff
@@ -0,0 +1,46 @@
+package rest
+
+import (
+	"errors"
+
+	"github.com/waku-org/go-waku/cmd/waku/server"
+	"github.com/waku-org/go-waku/waku/v2/protocol/pb"
+)
+
+type RestWakuMessage struct {
```

Comment:
I am little confused as to the original issue we faced which forced us to change this. 
Refer https://github.com/golang/protobuf/issues/1388#issuecomment-1644112541 
I looks like if either of the json tag matches, then the decoding should happen. 
Also, omitEmpty should have taken care of fields not being filled into json if they are empty. 

Wondering if we root-caused the issue correctly or missing something.

Reply:
Ah, and there is no way for us to specify to ignore these fields in protobuf generation.
Then i guess we don't have a choice.

Reply:
There's another option, we can use the protobuffer, and send an update to the waku-rest-api repo to include all the WakuMessage fields, including the rln proof!  I actually prefer that since it means less code for us to maintain

Reply:
Hmm, yeah..not a bad idea. We give all fields that are received and user can choose to ignore them if they don't want.